src/navigation/subNavigation/subNavigation.jsx

```diff
@@ -12,40 +39,70 @@ class SubNavigation extends React.Component {
     }
 
     componentDidUpdate(prevProps) {
-        if (this.props.selected !== prevProps.selected) {
-            this.setState({ selected: this.props.selected });
+        const {
+            selected,
+        } = this.props;
+
+        if (prevProps.selected !== selected) {
+            this.setState({ selected });
         }
     }
 
+    onItemClick(index, label, onChildClick, event) {
+        const {
+            onClick,
+        } = this.props;
+
+        if (isFunction(onClick)) {
+            onClick(index, label);
+        } else {
+            if (isFunction(onChildClick)) {
+                onChildClick(index, label);
+            }
+
+            this.setState({ selected: index });
+        }
+
+        event.preventDefault();
```

Comment:
Not sure if this is important, but usually you see `preventDefault()` or `stopPropagation()` near the top of the event handler function.  If this is working code, all code; no need to change!

Reply:
Fixed.

Reply:
Cool deal.  Resolving.